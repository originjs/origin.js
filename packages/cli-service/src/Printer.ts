import fse from 'fs-extra'
import type { InlineConfig } from 'vite'
import { baseConfig, buildConfig, serverConfig } from './config/base'
class Printer {
  name: string
  vueVersion: number
  rootDir: string
  css?: string[]
  typeScprict?: boolean
  test?: string[]
  eslint?: boolean
  localConfig?: InlineConfig

  constructor(dir: string) {
    const pkg = fse.readJsonSync(`${dir}/package.json`)
    this.rootDir = dir
    this.name = pkg.name
    this.vueVersion = 3
    if (fse.pathExistsSync(`${this.rootDir}/vite.config.js`))
      this.localConfig = require(`${this.rootDir}/vite.config.js`)
  }
  getBaseSchema(): InlineConfig {
    baseConfig.root = this.rootDir
    return baseConfig
  }
}

class DevPrinter extends Printer {
  constructor(dir: string) {
    super(dir)
  }
  getSchema() {
    const base = this.getBaseSchema()
    const _configs: InlineConfig = Object.assign(
      base,
      serverConfig,
      this.localConfig,
    )
    _configs.mode = 'development'
    return _configs
  }
}

class BuildPrinter extends Printer {
  constructor(dir: string) {
    super(dir)
  }
  getSchema() {
    const base = this.getBaseSchema()
    const _configs: InlineConfig = Object.assign(
      base,
      buildConfig,
      this.localConfig,
    )
    _configs.mode = 'production'
    return _configs
  }
}
export { DevPrinter, BuildPrinter }
