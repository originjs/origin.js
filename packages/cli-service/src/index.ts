import dev from './commands/dev'
import build from './commands/build'

export async function run(name: 'dev' | 'build') {
  if (name === 'dev') {
    await dev()
  } else if (name === 'build') {
    await build()
  }
}

export { dev, build }
