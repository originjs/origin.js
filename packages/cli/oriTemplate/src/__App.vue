<template>
  <main :class="isLight ? 'light' : 'dark'">
    <router-view />
    <div>
      <button
        type="button"
        @click="changeMode"
        class="normal-btn theme-ctrl-btn"
      >
        {{ !isLight ? $t('lightMode') : $t('darkMode') }}
      </button>
      <button
        type="button"
        @click="changeLocale"
        class="normal-btn theme-ctrl-btn"
      >
        {{ $t("changeLanguage") }}
      </button>
    </div>
  </main>
</template>

<script>
import { ref } from 'vue'
import { useI18n } from 'vue-i18n'

export default {
  name: 'App',
  setup: () => {
    const isLight = ref(true)

    const changeMode = () => {
      isLight.value = !isLight.value
    }

    const { locale } = useI18n()
    const changeLocale = () => {
      if (locale.value === 'en') {
        locale.value = 'zh'
      } else if (locale.value === 'zh') {
        locale.value = 'en'
      }
    }

    return { isLight, changeMode, changeLocale }
  }
}
</script>

<%_ if (!globalStylePluginImported) { _%>
<style src="./assets/global-theme.css"></style>
<%_ } _%>
<style scoped>
#app main {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  height: calc(100vh - 3.2rem);
  min-height: 568px;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
</style>
