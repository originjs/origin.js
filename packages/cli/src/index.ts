import { Command } from 'commander';
import { build } from './commands/build';
import { init } from './commands/init';
const program = new Command();
program.version(`@ori/cli ${require('../package.json').version}`)
    .usage('<command> [options]');

program.command('create <app-name>')
    .description('create a new project powered by ori-cli-service')
    .action((name, options) => {
        init(name, options)
})

program.command('buid <dir>')
    .description('Compiles the application for production deployment')
    .action((dir) => {
        build(dir);
    });

program.parse(process.argv);
