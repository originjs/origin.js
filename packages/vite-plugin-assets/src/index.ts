import path from 'path'
import fs from 'fs'
import { Plugin, HtmlTagDescriptor } from 'vite'

export type PluginOptions = {
  sourcePath?: string
  cssEnabled?: boolean
  sassEnabled?: boolean
  lessEnabled?: boolean
}

const DEFAULT_OPTIONS: PluginOptions = {
  sourcePath: '/src/assets',
  cssEnabled: true,
  sassEnabled: true,
  lessEnabled: true,
}

const GLOBALCSS = /^global-.*\.css$/
const GLOBALSASS = /^global-.*\.scss$/
const GLOBALLESS = /^global-.*\.less$/

//Get the path of the desired style sheet whose name starts with 'global-'
function searchGlobalCss(
  rootDir: string,
  options: PluginOptions,
): Array<string> {
  const globalCssPaths: Array<string> = []
  fs.readdirSync(rootDir).forEach(item => {
    const targetPath = path.resolve(rootDir, item)
    if (fs.statSync(targetPath).isDirectory()) {
      globalCssPaths.concat(searchGlobalCss(targetPath, options))
    } else {
      if (options.cssEnabled && GLOBALCSS.test(item)) {
        globalCssPaths.push(targetPath)
      }
      if (options.sassEnabled && GLOBALSASS.test(item)) {
        globalCssPaths.push(targetPath)
      }
      if (options.lessEnabled && GLOBALLESS.test(item)) {
        globalCssPaths.push(targetPath)
      }
    }
  })

  return globalCssPaths
}

export default (options: PluginOptions = {}): Plugin => {
  const opts: PluginOptions = Object.assign({}, DEFAULT_OPTIONS, options)

  return {
    name: 'vite:assets',
    transformIndexHtml: {
      enforce: 'pre',
      transform(
        _,
        { filename }: { filename: string },
      ): Array<HtmlTagDescriptor> {
        const HtmlTagDescriptors: Array<HtmlTagDescriptor> = []
        const assetsPath: string = path.resolve(
          filename,
          '..',
          opts.sourcePath!,
        )
        const filePaths: Array<string> = searchGlobalCss(assetsPath, opts)

        filePaths.forEach(filePath => {
          filePath = filePath
            .replace(assetsPath, opts.sourcePath!)
            .replace(/\\/g, '/')
          HtmlTagDescriptors.push({
            tag: 'link',
            attrs: {
              rel: 'stylesheet',
              href: filePath,
            },
            injectTo: 'head',
          })
        })

        return HtmlTagDescriptors
      },
    },
  }
}
