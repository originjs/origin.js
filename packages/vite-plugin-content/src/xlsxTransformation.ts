import { PluginOptions } from './index'
import { createFilter } from '@rollup/pluginutils'
import toSource from 'tosource'

export default function xlsxTransform(
  options: PluginOptions = {},
  code: string,
  id: string,
) {
  const xlsx = require('xlsx');
  const filter = createFilter(options.xlsx!.include, options.xlsx!.exclude)
  if (!filter(id)) {
    return null
  }

  const xlsxData = xlsx.readFile(id, options.xlsx!.xlsxOptions)
  const generatedCode = `var data = ${toSource(
    xlsxData,
  )};\nexport default data;`

  return {
    code: generatedCode,
  }
}
