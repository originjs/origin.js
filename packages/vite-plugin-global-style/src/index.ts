import path from 'path'
import fs from 'fs'
import { Plugin, HtmlTagDescriptor, HmrContext } from 'vite'

export type PluginOptions = {
  sourcePath: string
  cssEnabled: boolean
  sassEnabled: boolean
  lessEnabled: boolean
  stylusEnabled: boolean
  recursive: boolean
  debug: boolean
}

const DEFAULT_OPTIONS: PluginOptions = {
  sourcePath: 'src/assets',
  cssEnabled: true,
  sassEnabled: true,
  lessEnabled: true,
  stylusEnabled: true,
  recursive: true,
  debug: false,
}

type StyleData = {
  name: string
  globalRegex: RegExp
  extensionRegex: RegExp
  globalStylePaths: Array<string>
  isEnabled: (opt: PluginOptions) => boolean | undefined
}

const GLOBAL_STYLES_DATA: StyleData[] = [
  {
    name: 'css',
    globalRegex: /^global.*\.css$/,
    extensionRegex: /.css$/,
    globalStylePaths: [],
    isEnabled: (opt: PluginOptions) => {
      return opt.cssEnabled
    },
  },
  {
    name: 'sass',
    globalRegex: /^global.*\.scss$/,
    extensionRegex: /.scss(\?[\w]*)?$/,
    globalStylePaths: [],
    isEnabled: (opt: PluginOptions) => {
      return opt.sassEnabled
    },
  },
  {
    name: 'less',
    globalRegex: /^global.*\.less$/,
    extensionRegex: /.less(\?[\w]*)?$/,
    globalStylePaths: [],
    isEnabled: (opt: PluginOptions) => {
      return opt.lessEnabled
    },
  },
  {
    name: 'stylus',
    globalRegex: /^global.*\.styl(us)?$/,
    extensionRegex: /.styl(us)?(\?[\w]*)?$/,
    globalStylePaths: [],
    isEnabled: (opt: PluginOptions) => {
      return opt.stylusEnabled
    },
  },
]

//Get the path of the desired style sheet whose name starts with 'global'
function searchGlobalStyle(
  rootDir: string,
  options: PluginOptions,
  data: StyleData,
): Array<string> {
  let globalStylePaths: Array<string> = []

  fs.readdirSync(rootDir).forEach(item => {
    const targetPath = path.resolve(rootDir, item)
    if (fs.statSync(targetPath).isDirectory() && options.recursive) {
      globalStylePaths = globalStylePaths.concat(
        searchGlobalStyle(targetPath, options, data),
      )
    } else {
      if (data.globalRegex.test(item)) {
        globalStylePaths.push(targetPath)
      }
    }
  })

  return globalStylePaths
}

export default (options: Partial<PluginOptions> = {}): Plugin => {
  const opts: PluginOptions = Object.assign({}, DEFAULT_OPTIONS, options)
  let assetsPath: string

  return {
    name: 'vite:global-style',
    enforce: 'pre',
    transformIndexHtml: {
      enforce: 'pre',
      transform(
        _,
        { filename }: { filename: string },
      ): Array<HtmlTagDescriptor> {
        opts.debug && console.info(`[global style] filename: ${filename}`)
        const HtmlTagDescriptors: Array<HtmlTagDescriptor> = []
        if (!assetsPath) {
          assetsPath = path.resolve(filename, '..', opts.sourcePath)
          opts.debug && console.info(`[global style] root dir: ${assetsPath}`)
        }

        GLOBAL_STYLES_DATA.forEach(data => {
          data.globalStylePaths = searchGlobalStyle(assetsPath, opts, data)
          opts.debug &&
            console.info(
              `[global style] ${
                data.name
              } search result: \n${data.globalStylePaths.join('\n')}`,
            )
        })

        if (opts.cssEnabled) {
          // inject the global css styles to the index.html
          const cssStylesData = GLOBAL_STYLES_DATA.filter(value => {
            return value.name === 'css'
          })[0]
          const CSSFilePaths = cssStylesData.globalStylePaths

          CSSFilePaths.forEach(filePath => {
            filePath = filePath
              .replace(assetsPath, '/' + opts.sourcePath)
              .replace(/\\(\\)?/g, '/')
            HtmlTagDescriptors.push({
              tag: 'link',
              attrs: {
                rel: 'stylesheet',
                href: filePath,
              },
              injectTo: 'head',
            })
          })

          return HtmlTagDescriptors
        }

        return []
      },
    },
    transform(code: string, id: string) {
      let result = null
      GLOBAL_STYLES_DATA.forEach(data => {
        // the global css has already been handled in transformIndexHtml
        if (data.name === 'css') return

        if (
          data.extensionRegex.test(id) &&
          data.globalStylePaths.length > 0 &&
          data.isEnabled(opts)
        ) {
          // inject the global styles to corresponding files
          const globalImport = data.globalStylePaths
            .map(filePath => {
              filePath = path
                .relative(path.join(id, '..'), filePath)
                .replace(/\\(\\)?/g, '/')
              return `@import "${filePath}";`
            })
            .join(' ')
          result = {
            code: `${globalImport}\n${code}`,
            map: null,
          }
        }
      })

      return result
    },
    handleHotUpdate(ctx: HmrContext) {
      GLOBAL_STYLES_DATA.filter(data => {
        const baseName = path.basename(ctx.file)
        if (!data.globalRegex.test(baseName)) {
          return false
        }
        return true
      }).forEach(data => {
        // search again to update global style paths
        data.globalStylePaths = searchGlobalStyle(assetsPath, opts, data)
      })
    },
  }
}
