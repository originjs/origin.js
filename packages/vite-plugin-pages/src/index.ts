import { PluginOptions } from './types'
import { Plugin } from 'vite'
import { MODULE_NAME } from './constants'
import { generateCode, generateRoutes } from './generates'
import { getPages } from './pages'

export default (
  userOptions: PluginOptions = {
    root: process.cwd(),
    pagesDir: 'src/pages',
    layoutsDir: 'src/layouts',
    extensions: ['vue'],
  },
): Plugin => {
  const options: PluginOptions = Object.assign({}, userOptions)

  return {
    name: 'vite:pages',
    enforce: 'pre',
    resolveId(id) {
      if (id === MODULE_NAME) {
        return id
      }
      return null
    },
    async load(id) {
      if (id !== MODULE_NAME) {
        return
      }
      const pages = getPages(options.pagesDir, options.extensions)
      const routes = generateRoutes(pages, options)
      return generateCode(routes)
    },
    async transform(_code, id) {
      if (!/vue&type=layout/.test(id)) {
        return
      }
      return {
        code: 'export default {}',
        map: null,
      }
    },
  }
}

export { PluginOptions }
