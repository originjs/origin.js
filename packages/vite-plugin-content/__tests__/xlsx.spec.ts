import xlsxTransform from '../src/xlsxTransformation'
import { join } from 'path'
import fs from 'fs'
import { PluginOptions } from '../src'

describe('xlsxTest', () => {
  test('include and exclude test', async () => {
    let options: PluginOptions = {
      xlsx: {
        enabled: true,
        include: /testInclude.xlsx/,
        exclude: /testEXCEL.xlsx/,
      },
    }

    const path: string = join(__dirname, './files/testEXCEL.xlsx')
    let transformResult = xlsxTransform(options, '', path)
    expect(transformResult).toBeNull()

    transformResult = xlsxTransform(options, '', path)
    expect(transformResult).toBeNull()

    options = {
      xlsx: {
        enabled: true,
        include: /testEXCEL.xlsx/,
      },
    }
    transformResult = xlsxTransform(options, '', path)
    expect(transformResult).not.toBeNull()
  })

  test('xlsx transform', async () => {
    const path: string = join(__dirname, './files/testEXCEL.xlsx')
    const xlsxContent: string = fs.readFileSync(path).toString()
    const options: PluginOptions = { xlsx: { enabled: true } }
    const transformResult = xlsxTransform(options, xlsxContent, path)
    expect(transformResult!.code).toMatchSnapshot('A1')
  })
})
