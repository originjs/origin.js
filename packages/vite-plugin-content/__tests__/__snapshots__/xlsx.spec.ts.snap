// Vitest Snapshot v1

exports[`xlsxTest > xlsx transform > A1 1`] = `
"var data = { Directory:{ workbooks:[ \\"/xl/workbook.xml\\" ],
    sheets:[ \\"/xl/worksheets/sheet2.xml\\",
      \\"/xl/worksheets/sheet3.xml\\",
      \\"/xl/worksheets/sheet1.xml\\" ],
    charts:[  ],
    dialogs:[  ],
    macros:[  ],
    rels:[  ],
    strs:[ \\"/xl/sharedStrings.xml\\" ],
    comments:[  ],
    links:[  ],
    coreprops:[ \\"/docProps/core.xml\\" ],
    extprops:[ \\"/docProps/app.xml\\" ],
    custprops:[  ],
    themes:[ \\"/xl/theme/theme1.xml\\" ],
    styles:[ \\"/xl/styles.xml\\" ],
    vba:[  ],
    drawings:[  ],
    TODO:[  ],
    xmlns:\\"http://schemas.openxmlformats.org/package/2006/content-types\\",
    calcchain:\\"/xl/calcChain.xml\\",
    sst:\\"/xl/sharedStrings.xml\\",
    style:\\"/xl/styles.xml\\",
    defaults:{ rels:\\"application/vnd.openxmlformats-package.relationships+xml\\",
      xml:\\"application/xml\\" } },
  Workbook:{ AppVersion:{ appName:\\"xl\\",
      appname:\\"xl\\",
      lastEdited:\\"4\\",
      lastedited:\\"4\\",
      lowestEdited:\\"4\\",
      lowestedited:\\"4\\",
      rupBuild:\\"4505\\",
      rupbuild:\\"4505\\" },
    WBProps:{ defaultThemeVersion:124226,
      allowRefreshQuery:false,
      autoCompressPictures:true,
      backupFile:false,
      checkCompatibility:false,
      CodeName:\\"\\",
      date1904:false,
      filterPrivacy:false,
      hidePivotFieldList:false,
      promptedSolutions:false,
      publishItems:false,
      refreshAllConnections:false,
      saveExternalLinkValues:true,
      showBorderUnselectedTables:true,
      showInkAnnotation:true,
      showObjects:\\"all\\",
      showPivotChartFilter:false,
      updateLinks:\\"userSet\\" },
    WBView:[ { xWindow:\\"0\\",
        xwindow:\\"0\\",
        yWindow:\\"1170\\",
        ywindow:\\"1170\\",
        windowWidth:\\"10365\\",
        windowwidth:\\"10365\\",
        windowHeight:\\"8970\\",
        windowheight:\\"8970\\",
        activeTab:0,
        autoFilterDateGrouping:true,
        firstSheet:0,
        minimized:false,
        showHorizontalScroll:true,
        showSheetTabs:true,
        showVerticalScroll:true,
        tabRatio:600,
        visibility:\\"visible\\" } ],
    Sheets:[ { name:\\"Feuil1\\",
        sheetId:\\"1\\",
        sheetid:\\"1\\",
        id:\\"rId1\\",
        Hidden:0 },
      { name:\\"Feuil2\\",
        sheetId:\\"2\\",
        sheetid:\\"2\\",
        id:\\"rId2\\",
        Hidden:0 },
      { name:\\"Feuil3\\",
        sheetId:\\"3\\",
        sheetid:\\"3\\",
        id:\\"rId3\\",
        Hidden:0 } ],
    CalcPr:{ calcId:\\"125725\\",
      calcid:\\"125725\\",
      calcCompleted:\\"true\\",
      calcMode:\\"auto\\",
      calcOnSave:\\"true\\",
      concurrentCalc:\\"true\\",
      fullCalcOnLoad:\\"false\\",
      fullPrecision:\\"true\\",
      iterate:\\"false\\",
      iterateCount:\\"100\\",
      iterateDelta:\\"0.001\\",
      refMode:\\"A1\\" },
    Names:[  ],
    xmlns:\\"http://schemas.openxmlformats.org/spreadsheetml/2006/main\\",
    Views:[ {} ] },
  Props:{ LastAuthor:\\"garribas\\",
    Author:\\"Keith Bennett\\",
    Title:\\"Simple Excel document\\",
    CreatedDate:new Date(1191255236000),
    ModifiedDate:new Date(1229011337000),
    Application:\\"Microsoft Excel\\",
    AppVersion:\\"12.0000\\",
    DocSecurity:\\"0\\",
    HyperlinksChanged:false,
    SharedDoc:false,
    LinksUpToDate:false,
    ScaleCrop:false,
    Worksheets:3,
    SheetNames:[ \\"Feuil1\\",
      \\"Feuil2\\",
      \\"Feuil3\\" ] },
  Custprops:{},
  Deps:{},
  Sheets:{ Feuil1:{ \\"!ref\\":\\"A1:D22\\",
      A1:{ t:\\"s\\",
        v:\\"Sample Excel Worksheet - Numbers and their Squares\\",
        r:\\"<t>Sample Excel Worksheet - Numbers and their Squares</t>\\",
        h:\\"Sample Excel Worksheet - Numbers and their Squares\\",
        w:\\"Sample Excel Worksheet - Numbers and their Squares\\" },
      B4:{ t:\\"s\\",
        v:\\"Number\\",
        r:\\"<t>Number</t>\\",
        h:\\"Number\\",
        w:\\"Number\\" },
      C4:{ t:\\"s\\",
        v:\\"Square\\",
        r:\\"<t>Square</t>\\",
        h:\\"Square\\",
        w:\\"Square\\" },
      B6:{ t:\\"n\\",
        v:1,
        w:\\"1\\" },
      C6:{ t:\\"n\\",
        v:1,
        f:\\"+B6*B6\\",
        w:\\"1\\" },
      B7:{ t:\\"n\\",
        v:2,
        f:\\"+B6+1\\",
        w:\\"2\\" },
      C7:{ t:\\"n\\",
        v:4,
        f:\\"+B7*B7\\",
        w:\\"4\\" },
      B8:{ t:\\"n\\",
        v:3,
        f:\\"+B7+1\\",
        w:\\"3\\" },
      C8:{ t:\\"n\\",
        v:9,
        f:\\"+B8*B8\\",
        w:\\"9\\" },
      B9:{ t:\\"n\\",
        v:4,
        f:\\"+B8+1\\",
        w:\\"4\\" },
      C9:{ t:\\"n\\",
        v:16,
        f:\\"+B9*B9\\",
        w:\\"16\\" },
      B10:{ t:\\"n\\",
        v:5,
        f:\\"+B9+1\\",
        w:\\"5\\" },
      C10:{ t:\\"n\\",
        v:25,
        f:\\"+B10*B10\\",
        w:\\"25\\" },
      B11:{ t:\\"n\\",
        v:6,
        f:\\"+B10+1\\",
        w:\\"6\\" },
      C11:{ t:\\"n\\",
        v:36,
        f:\\"+B11*B11\\",
        w:\\"36\\" },
      B12:{ t:\\"n\\",
        v:7,
        f:\\"+B11+1\\",
        w:\\"7\\" },
      C12:{ t:\\"n\\",
        v:49,
        f:\\"+B12*B12\\",
        w:\\"49\\" },
      B13:{ t:\\"n\\",
        v:8,
        f:\\"+B12+1\\",
        w:\\"8\\" },
      C13:{ t:\\"n\\",
        v:64,
        f:\\"+B13*B13\\",
        w:\\"64\\" },
      B14:{ t:\\"n\\",
        v:9,
        f:\\"+B13+1\\",
        w:\\"9\\" },
      C14:{ t:\\"n\\",
        v:81,
        f:\\"+B14*B14\\",
        w:\\"81\\" },
      B15:{ t:\\"n\\",
        v:10,
        f:\\"+B14+1\\",
        w:\\"10\\" },
      C15:{ t:\\"n\\",
        v:100,
        f:\\"+B15*B15\\",
        w:\\"100\\" },
      B16:{ t:\\"n\\",
        v:11,
        f:\\"+B15+1\\",
        w:\\"11\\" },
      C16:{ t:\\"n\\",
        v:121,
        f:\\"+B16*B16\\",
        w:\\"121\\" },
      B17:{ t:\\"n\\",
        v:12,
        f:\\"+B16+1\\",
        w:\\"12\\" },
      C17:{ t:\\"n\\",
        v:144,
        f:\\"+B17*B17\\",
        w:\\"144\\" },
      B18:{ t:\\"n\\",
        v:13,
        f:\\"+B17+1\\",
        w:\\"13\\" },
      C18:{ t:\\"n\\",
        v:169,
        f:\\"+B18*B18\\",
        w:\\"169\\" },
      B19:{ t:\\"n\\",
        v:14,
        f:\\"+B18+1\\",
        w:\\"14\\" },
      C19:{ t:\\"n\\",
        v:196,
        f:\\"+B19*B19\\",
        w:\\"196\\" },
      B20:{ t:\\"n\\",
        v:15,
        f:\\"+B19+1\\",
        w:\\"15\\" },
      C20:{ t:\\"n\\",
        v:225,
        f:\\"+B20*B20\\",
        w:\\"225\\" },
      A22:{ t:\\"s\\",
        v:\\"Written and saved in Microsoft Excel X for Mac Service Release 1.\\\\r\\\\n\\",
        r:\\"<t xml:space=\\\\\\"preserve\\\\\\">Written and saved in Microsoft Excel X for Mac Service Release 1.\\\\r\\\\n</t>\\",
        h:\\"Written and saved in Microsoft Excel X for Mac Service Release 1.&#x000d;<br/>\\",
        w:\\"Written and saved in Microsoft Excel X for Mac Service Release 1.\\\\r\\\\n\\" },
      \\"!margins\\":{ left:0.75,
        right:0.75,
        top:1,
        bottom:1,
        header:0.4921259845,
        footer:0.4921259845 } },
    Feuil2:{ \\"!margins\\":{ left:0.75,
        right:0.75,
        top:1,
        bottom:1,
        header:0.4921259845,
        footer:0.4921259845 } },
    Feuil3:{ \\"!margins\\":{ left:0.75,
        right:0.75,
        top:1,
        bottom:1,
        header:0.4921259845,
        footer:0.4921259845 } } },
  SheetNames:[ \\"Feuil1\\",
    \\"Feuil2\\",
    \\"Feuil3\\" ],
  Strings:[ { t:\\"Sample Excel Worksheet - Numbers and their Squares\\",
      r:\\"<t>Sample Excel Worksheet - Numbers and their Squares</t>\\",
      h:\\"Sample Excel Worksheet - Numbers and their Squares\\" },
    { t:\\"Number\\",
      r:\\"<t>Number</t>\\",
      h:\\"Number\\" },
    { t:\\"Square\\",
      r:\\"<t>Square</t>\\",
      h:\\"Square\\" },
    { t:\\"Written and saved in Microsoft Excel X for Mac Service Release 1.\\\\r\\\\n\\",
      r:\\"<t xml:space=\\\\\\"preserve\\\\\\">Written and saved in Microsoft Excel X for Mac Service Release 1.\\\\r\\\\n</t>\\",
      h:\\"Written and saved in Microsoft Excel X for Mac Service Release 1.&#x000d;<br/>\\" },
    { t:\\"\\" } ],
  Styles:{ Fonts:[ { sz:10,
        name:\\"Arial\\" },
      { bold:1,
        underline:1,
        sz:10,
        name:\\"Verdana\\" } ],
    Fills:[ { patternType:\\"none\\" },
      { patternType:\\"gray125\\" } ],
    Borders:[ {} ],
    CellXf:[ { numFmtId:0,
        numfmtid:\\"0\\",
        fontId:0,
        fontid:\\"0\\",
        fillId:0,
        fillid:\\"0\\",
        borderId:0,
        borderid:\\"0\\",
        xfId:0,
        xfid:\\"0\\" },
      { numFmtId:0,
        numfmtid:\\"0\\",
        fontId:1,
        fontid:\\"1\\",
        fillId:0,
        fillid:\\"0\\",
        borderId:0,
        borderid:\\"0\\",
        xfId:0,
        xfid:\\"0\\",
        applyFont:true,
        applyfont:\\"1\\" },
      { numFmtId:0,
        numfmtid:\\"0\\",
        fontId:1,
        fontid:\\"1\\",
        fillId:0,
        fillid:\\"0\\",
        borderId:0,
        borderid:\\"0\\",
        xfId:0,
        xfid:\\"0\\",
        applyFont:true,
        applyfont:\\"1\\",
        applyAlignment:true,
        applyalignment:\\"1\\",
        alignment:{ horizontal:\\"right\\" } },
      { numFmtId:0,
        numfmtid:\\"0\\",
        fontId:0,
        fontid:\\"0\\",
        fillId:0,
        fillid:\\"0\\",
        borderId:0,
        borderid:\\"0\\",
        xfId:0,
        xfid:\\"0\\",
        applyAlignment:true,
        applyalignment:\\"1\\",
        alignment:{ wrapText:true } } ] },
  Themes:{},
  SSF:{ 0:\\"General\\",
    1:\\"0\\",
    2:\\"0.00\\",
    3:\\"#,##0\\",
    4:\\"#,##0.00\\",
    9:\\"0%\\",
    10:\\"0.00%\\",
    11:\\"0.00E+00\\",
    12:\\"# ?/?\\",
    13:\\"# ??/??\\",
    14:\\"m/d/yy\\",
    15:\\"d-mmm-yy\\",
    16:\\"d-mmm\\",
    17:\\"mmm-yy\\",
    18:\\"h:mm AM/PM\\",
    19:\\"h:mm:ss AM/PM\\",
    20:\\"h:mm\\",
    21:\\"h:mm:ss\\",
    22:\\"m/d/yy h:mm\\",
    37:\\"#,##0 ;(#,##0)\\",
    38:\\"#,##0 ;[Red](#,##0)\\",
    39:\\"#,##0.00;(#,##0.00)\\",
    40:\\"#,##0.00;[Red](#,##0.00)\\",
    45:\\"mm:ss\\",
    46:\\"[h]:mm:ss\\",
    47:\\"mmss.0\\",
    48:\\"##0.0E+0\\",
    49:\\"@\\",
    56:\\"\\\\\\"上午/下午 \\\\\\"hh\\\\\\"時\\\\\\"mm\\\\\\"分\\\\\\"ss\\\\\\"秒 \\\\\\"\\" } };
export default data;"
`;
